usim/_primitives/context.py

```diff
@@ -265,13 +305,11 @@ def _collect_exceptions(self)\
         suppress = self.SUPPRESS_CONCURRENT
         promote = self.PROMOTE_CONCURRENT
         concurrent = []
-        for child in self._children:
-            if child.status is TaskState.FAILED:
-                exc = child.__exception__
-                if isinstance(exc, promote):
-                    return exc, None
-                if not isinstance(exc, suppress):
-                    concurrent.append(exc)
+        for exc in self._child_failures:
```

Comment:
I really like the collection of `_child_failures` when children actually finish. You are saving further checks and children can be cleaned up early.

Reply:
Thanks!